v2/jdbc-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/JdbcToBigQuery.java

```diff
@@ -160,21 +165,53 @@ static PipelineResult run(JdbcToBigQueryOptions options, Write<TableRow> writeTo
     PCollection<TableRow> rows;
     if (options.getPartitionColumn() != null && options.getTable() != null) {
       // Read with Partitions
-      // TODO(pranavbhandari): Support readWithPartitions for other data types.
-      JdbcIO.ReadWithPartitions<TableRow, Long> readIO =
-          JdbcIO.<TableRow>readWithPartitions()
+      JdbcIO.ReadWithPartitions<TableRow, ?> readIO = null;
+      final String partitionColumntype = options.getPartitionColumnType();
+      if(partitionColumntype == null ){ 
+        throw new IllegalStateException("Expected non-null partitionColumnType");
+      } else if("long".equals(partitionColumntype)){
+        JdbcIO.ReadWithPartitions<TableRow, Long> longTypeReadIO = 
+            JdbcIO.<TableRow,Long>readWithPartitions(TypeDescriptors.longs())
               .withDataSourceConfiguration(dataSourceConfiguration)
               .withTable(options.getTable())
               .withPartitionColumn(options.getPartitionColumn())
               .withRowMapper(JdbcConverters.getResultSetToTableRow(options.getUseColumnAlias()));
+        if (options.getLowerBound() != null && options.getUpperBound() != null) {
+          // Check if lower bound and upper bound are long type.
+          try {
+            longTypeReadIO =
+              longTypeReadIO.withLowerBound(Long.valueOf(options.getLowerBound())).withUpperBound(Long.valueOf(options.getUpperBound()));
+            }
+          catch (NumberFormatException e){
+              throw new NumberFormatException("Expected Long values for lowerBound and upperBound, received : " + e.getMessage());
+          }
+        }
+      readIO = longTypeReadIO;
+      } else if ("datetime".equals(partitionColumntype)){
+        JdbcIO.ReadWithPartitions<TableRow, DateTime> dateTimeReadIO =  
+            JdbcIO.<TableRow,DateTime>readWithPartitions(TypeDescriptor.of(DateTime.class))
+              .withDataSourceConfiguration(dataSourceConfiguration)
+              .withTable(options.getTable())
+              .withPartitionColumn(options.getPartitionColumn())
+              .withRowMapper(JdbcConverters.getResultSetToTableRow(options.getUseColumnAlias()));
+          if (options.getLowerBound() != null && options.getUpperBound() != null) {
+            DateTimeFormatter dateFormatter = DateTimeFormat.forPattern("YYYY-MM-dd HH:mm:ss");
```

Comment:
this pattern should be documented in options, especially if documentation is generated based on it.

Reply:
probably timezone should be included, because it is a usual problem when not handled :(https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_OFFSET_DATE_TIME)

ex: BigQuery is also showing the datetime in similar manner : https://cloud.google.com/bigquery/docs/reference/standard-sql/datetime_functions#datetime